bridges/snowbridge/pallets/outbound-queue-v2/src/send_message_impl.rs

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+//! Implementation for [`snowbridge_outbound_queue_primitives::v2::SendMessage`]
+use super::*;
+use bridge_hub_common::AggregateMessageOrigin;
+use codec::Encode;
+use frame_support::{
+	ensure,
+	traits::{EnqueueMessage, Get},
+};
+use snowbridge_outbound_queue_primitives::{
+	v2::{primary_governance_origin, Message, SendMessage},
+	SendError, SendMessageFeeProvider,
+};
+use sp_core::H256;
+use sp_runtime::BoundedVec;
+
+impl<T> SendMessage for Pallet<T>
+where
+	T: Config,
+{
+	type Ticket = Message;
+
+	type Balance = T::Balance;
+
+	fn validate(message: &Message) -> Result<(Self::Ticket, Self::Balance), SendError> {
+		// The inner payload should not be too large
+		let payload = message.encode();
+		ensure!(
+			payload.len() < T::MaxMessagePayloadSize::get() as usize,
+			SendError::MessageTooLarge
+		);
+
+		let fee = Self::calculate_local_fee();
+
+		Ok((message.clone(), fee))
+	}
+
+	fn deliver(ticket: Self::Ticket) -> Result<H256, SendError> {
+		let origin = AggregateMessageOrigin::SnowbridgeV2(ticket.origin);
+
+		if ticket.origin != primary_governance_origin() {
+			ensure!(!Self::operating_mode().is_halted(), SendError::Halted);
+		}
```

Comment:
@yrong 

I think we should rather halt outbound transfers in AssetHub, otherwise users would lose their funds.

Since we now have a system pallet on AssetHub, we can add a `halt()` governance extrinsic that tells the XCM Exporter to stop sending messages to BH. With this approach, when the transfer is halted on AH, the funds are trapped safely.

And governance extrinsics on BH would still be allowed to send messages to Ethereum

Reply:
^ @acatangiu does this make sense to you?

Reply:
> add a halt() governance extrinsic that tells the XCM Exporter

I would assume the halt call will set a Boolean flag in runtime storage,  and we may need a custom exporter to respect that Boolean flag.

Reply:
https://github.com/yrong/polkadot-sdk/pull/12